include/swift/SIL/SILOpenedArchetypesTracker.h

```diff
@@ -20,6 +20,12 @@
 
 namespace swift {
 
+// Disable MSVC warning: multiple copy constructors specified.
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable: 4521)
+#endif
```

Comment:
I'm inclined to think this is an MSVC bug. I think it is related to 
 The line `SILOpenedArchetypesTracker &operator = (const SILOpenedArchetypesTracker &) = delete;` but have been unable to put a repo together.

I'm not surre I fully understand the `enable_if` comment you made, sorry